mlflow/utils/checkpoint_utils.py

```
@@ -0,0 +1,174 @@
+            else:
+                checkpoint_model_filename = f"checkpoint.{suffix}"
+            checkpoint_metrics_filename = "checkpoint_metrics.json"
+            checkpoint_artifact_dir = f"checkpoints/{sub_dir_name}"
+
+        mlflow.set_tag(
+            LATEST_CHECKPOINT_ARTIFACT_TAG_KEY,
+            f"{checkpoint_artifact_dir}/{checkpoint_model_filename}",
+        )
+
+        mlflow.log_dict(
+            {**metric_dict, "epoch": current_epoch, "global_step": global_step},
+            f"{checkpoint_artifact_dir}/{checkpoint_metrics_filename}",
+        )
+
+        tmp_dir = create_tmp_dir()
+        try:
+            tmp_model_save_path = os.path.join(tmp_dir, checkpoint_model_filename)
+            self.save_checkpoint(tmp_model_save_path)
+            mlflow.log_artifact(tmp_model_save_path, checkpoint_artifact_dir)
+        finally:
+            shutil.rmtree(tmp_dir, ignore_errors=True)
+
+
+def download_checkpoint_artifact(run_id=None, epoch=None, global_step=None):
+    from mlflow.utils.mlflow_tags import LATEST_CHECKPOINT_ARTIFACT_TAG_KEY
+    from mlflow.client import MlflowClient
+
+    client = MlflowClient()
+
+    if run_id is None:
+        run = mlflow.active_run()
+        if run is None:
+            raise MlflowException(
+                "There is no active run, please provide the 'run_id' for " "'load_checkpoint' call."
+            )
+        run_id = run.info.run_id
+    else:
+        run = client.get_run(run_id)
+
+    latest_checkpoint_artifact_path = run.data.tags.get(LATEST_CHECKPOINT_ARTIFACT_TAG_KEY)
+    if latest_checkpoint_artifact_path is None:
+        raise MlflowException("There is no logged checkpoint artifact in the current run.")
+
+    checkpoint_filename = os.path.basename(latest_checkpoint_artifact_path)
+
+    if epoch is not None and global_step is not None:
+        raise MlflowException(
+            "Only one of 'epoch' and 'global_step' can be set for 'load_checkpoint'."
+        )
```

We can throw this exception earlier before fetching a run.